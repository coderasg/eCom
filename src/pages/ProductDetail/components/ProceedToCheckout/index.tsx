import React, { useCallback } from 'react';

import Button from './../../../../components/Button';

import  styles from './proceedToCheckout.module.scss';

type Props = {
  price: number;
  shippingFee: string;
}

function ProceedToCheckout(props: Props) {
  const { price, shippingFee } = props;

  const handleCheckoutClick = useCallback(() => {
    alert('Proceeding Checkout');
  }, []);

  return <div data-testid="ProceedToCheckout-wrapper" className={styles.proceedToCheckout}>
    <div><span className={styles.price}>{`Ò°${price}`}</span> <span className={styles.shippinFee}>{shippingFee}</span></div>
    <Button className={styles.button}  onClick={handleCheckoutClick}>{'Proceed To Checkout'}</Button>
  </div>;
}

export default ProceedToCheckout;
