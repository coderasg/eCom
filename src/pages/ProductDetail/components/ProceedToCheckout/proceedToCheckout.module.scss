@use './../../../../variables';

.proceedToCheckout {
  position: fixed;
  bottom: 0px;
  padding: variables.$paddingMid;
  width: 100%;
  background-color: variables.$footerColor;
  color: variables.$white;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .price {
    font-size: 20px;
  }

  .shippinFee {
    font-size: 10px;
  }
}
