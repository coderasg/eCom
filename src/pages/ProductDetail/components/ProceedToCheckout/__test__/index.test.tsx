import React from 'react';
import renderer from 'react-test-renderer';

import ProceedToCheckout from '../index';

describe('(Component) ProceedToCheckout', () => {
  const defaultProps = {
    price: 1332,
    shippingFee: 'included'
  };
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<ProceedToCheckout {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
