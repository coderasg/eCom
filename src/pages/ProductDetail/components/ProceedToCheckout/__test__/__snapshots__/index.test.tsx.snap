// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProceedToCheckout should pass snapshot test 1`] = `
<div
  data-testid="ProceedToCheckout-wrapper"
>
  <div>
    <span>
      Ò°1332
    </span>
     
    <span>
      included
    </span>
  </div>
  <button
    className="undefined undefined"
    data-testid="Button-wrapper"
    onClick={[Function]}
  >
    Proceed To Checkout
  </button>
</div>
`;
